src/nft/components/collection/CollectionNfts.tsx

```diff
@@ -287,6 +288,72 @@ export const CollectionNfts = ({ contractAddress, collectionStats, rarityVerifie
 
   const { assets: collectionNfts, loadNext, hasNext, isLoadingNext } = useLazyLoadAssetsQuery(assetQueryParams)
 
+  const isSelected = useCallback(
+    (asset: GenieAsset) => {
+      return itemsInBag.some((item) => asset.tokenId === item.asset.tokenId && asset.address === item.asset.address)
+    },
+    [itemsInBag]
+  )
+
+  const calculatePrice = useCallback(
+    (asset: GenieAsset) => {
+      return isSelected(asset)
+        ? calcPoolPrice(
+            asset,
+            itemsInBag
+              .filter((item) => item.asset.address === asset.address && item.asset.marketplace === asset.marketplace)
+              .map((item) => item.asset.tokenId)
+              .indexOf(asset.tokenId)
+          )
+        : calcPoolPrice(
+            asset,
+            itemsInBag.filter(
+              (item) => item.asset.address === asset.address && item.asset.marketplace === asset.marketplace
+            ).length
+          )
+    },
+    [isSelected, itemsInBag]
+  )
+
+  const collectionAssets = useMemo(() => {
+    if (
+      !collectionNfts ||
+      !collectionNfts.some((asset) => asset.marketplace === Markets.NFTX || asset.marketplace === Markets.NFT20)
+    ) {
+      return collectionNfts
+    }
+
+    const collectionAssets = [...collectionNfts]
+
+    collectionAssets.forEach(
+      (asset) =>
+        (asset.marketplace === Markets.NFTX || asset.marketplace === Markets.NFT20) &&
+        (asset.priceInfo.ETHPrice = calculatePrice(asset))
+    )
+
+    if (sortBy === SortBy.HighToLow || sortBy === SortBy.LowToHigh) {
+      collectionAssets.sort((a, b) => {
+        const bigA = BigNumber.from(a.priceInfo.ETHPrice ?? -1)
+        const bigB = BigNumber.from(b.priceInfo.ETHPrice ?? -1)
+        const diff = bigA.sub(bigB)
+
+        if ((bigA.gte(0) || bigB.gte(0)) && (bigA.lt(0) || bigB.lt(0))) {
+          return bigA.gte(0) ? (sortBy === SortBy.LowToHigh ? -1 : 1) : sortBy === SortBy.LowToHigh ? 1 : -1
```

Comment:
nit: unpack this compound ternary into nest if/else so it's a little easier to follow (I like lines 344-348 below--that could be written this way, but it's easier to follow w/ if/elseIf and return <single ternary expression>